src/Adapter/Swoole/Internal/DummyPromise.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace M6Web\Tornado\Adapter\Swoole\Internal;
+
+use M6Web\Tornado\Promise;
+use M6Web\Tornado\Deferred;
+use Throwable;
+
+final class DummyPromise implements Promise, Deferred
+{
+    private bool $isPending;
+    private mixed $value;
+    private ?Throwable $exception;
+    private $callback;
+
+    public function __construct(?callable $callback = null)
+    {
+        $this->isPending = true;
+        $this->exception = null;
+        $this->callback = $callback ?? static function() {};
+    }
+    
+    public static function wrap(Promise $promise): self
+    {
+        assert($promise instanceof self, new \Error('Input promise was not created by this adapter.'));
+
+        return $promise;
+    }
+    
+    public function getPromise(): Promise
+    {
+        return $this;
+    }
+
+    public function resolve($value): void
+    {
+        assert(true === $this->isPending, new \Error('Promise is already resolved.'));
+
+        $this->isPending = false;
+        $this->value = $value;
+        ($this->callback)();
+    }
+
+    public function reject(Throwable $throwable): void
+    {
+        assert(true === $this->isPending, new \Error('Promise is already resolved.'));
+
+        $this->isPending = false;
+        $this->exception = $throwable;
+        ($this->callback)();
+    }
+
+    public function isPending(): bool
+    {
+        if($this->isPending) {
+            return $this->isPending;
+        }
+
+        if($this->exception === null) {
+            if ($this->value instanceof self) {
```

Comment:
I don't get it… why the value of a promise would be a promise? An why does it change something? 🤔

Reply:
This was add to fullfill needs for `eventloop->resolve` that is involved in `wait` and `async`
`DummyPromise` is much used here as a container and all logic for Swoole is implemented in `EventLoop`.
In fact, I could have written `isPending` logic to a new private method in `EventLoop->isPending($promise)`.